### ISSUE TITLE
If it is a 16-line lidar, what parameters need to be modified?

### ISSUE THREAD

**Issue by hunkyu** [irapkaist/removert#2]
If it is a 16-line lidar, what parameters need to be modified?

**Comment by gisbi-kim** [comment]
Our algorithm requires not only number of rays, but also the vertical for in degree (when the lidar sensor is the origin)

you can see this concept at this line
https://github.com/irapkaist/removert/blob/65566bb2dc5586020839fc11a925765b0c72399d/src/Removerter.cpp#L199

in summary, 
you need to change the variable "rimg_shape" and "kVFOV"

for example, if you are using a 16 ray lidar having [-25 deg, +25 deg] vertical fov, 
you can just use kVFOV = (25)-(-25) = 50 (as already used in the code)

but we suggest to adjust the range image shape (i.e., a size of a single pixel in a range image).
for the 16 ray lidar, with naive assumption, a single ray is corresponding 50/16 ~ 3 deg
for this case, we recommend to start to remove with a range image size whose pixel size is corresponding 3deg x 3deg

In our code, the rimg_shape is adjusted using a single parameter _res_alpha)
Please refer this function
https://github.com/irapkaist/removert/blob/65566bb2dc5586020839fc11a925765b0c72399d/src/utility.cpp#L54
That is, if you use alpha = 1, then a single pixel for 1 deg x 1deg, or if you use alpha = 0.5, then a single pixel for 2 deg x 2 deg.
this empirical guideline was summarized in this line in the config yaml file 
https://github.com/irapkaist/removert/blob/65566bb2dc5586020839fc11a925765b0c72399d/config/params.yaml#L60

In conclusion, 
Thus you need to check your Lidar’s vertical fov range as well as the number of rays.

Thank you and enjoy.

**Comment by hunkyu** [comment]
Well,I see!

**Comment by hunkyu** [comment]
Does this go to the Ladar's own coordinate system? What to do if it is a car body coordinate system?

**Comment by gisbi-kim** [comment]
@hunkyu 
The lidar own coordinate-based se3 trajectory is required. 
you can make it easily by existing lidar odometry methods 
e.g., https://github.com/gisbi-kim/SC-LIO-SAM#applications
